### PR TITLE
Added basic census schemas.

### PR DESCRIPTION
**Pull request by jonnyshaw89**
### What is the context of this PR?
Added basic census schemas.
Mainly Household survey but includes skeleton schemas for individual and communal establishment.

### CODE REVIEW

File: tests/functional/generate_pages.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+def process_block(block, dir_out, spec_out):
+    logger.debug("Processing Block: %s", block['id'])
+
+    page_filename = block['id'] + '.page.js'
+    page_path = os.path.join(dir_out, page_filename)
+
+    logger.info("Creating %s...", page_path)
+
+    with open(page_path, 'w') as page_spec:
+        radio_check = find_kv(block, 'type', 'Radio')
```
Comment (collisdigital) [inline tests/functional/generate_pages.py]: Does this also need to check for Checkbox types?

File: tests/functional/generate_pages.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+        option_name = generate_camel_case_from_id(option['value'])
+        option_id = "{name}-{index}".format(name=answer_id, index=index+1)
+        page_spec.write(CHECKBOX_RADIO_CLICKER.replace("{optionName}", option_name).replace("{optionId}", option_id))
+
+
+def process_answer(answer, page_spec):
+    if answer['type'] == 'Radio' or answer['type'] == 'Checkbox':
+        process_options(answer['id'], answer['options'], page_spec)
+
+    if answer['type'] == 'Date':
```
Comment (collisdigital) [inline tests/functional/generate_pages.py]: can be `elif`

File: tests/functional/generate_pages.py [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+        answer_name = generate_camel_case_from_id(answer['id'])
```
Comment (collisdigital) [inline tests/functional/generate_pages.py:86]: Worth moving all this to a separate function to keep it clean?

File: tests/functional/generate_pages.py [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        page_spec.write(ANSWER_SETTER.replace("{answerName}", answer_name + 'Day').replace("{answerId}", answer['id'] + '-day'))
+        page_spec.write(ANSWER_GETTER.replace("{answerName}", answer_name + 'Day').replace("{answerId}", answer['id'] + '-day'))
+        page_spec.write(ANSWER_SETTER.replace("{answerName}", answer_name + 'Month').replace("{answerId}", answer['id'] + '-month'))
+        page_spec.write(ANSWER_GETTER.replace("{answerName}", answer_name + 'Month').replace("{answerId}", answer['id'] + '-month'))
+        page_spec.write(ANSWER_SETTER.replace("{answerName}", answer_name + 'Year').replace("{answerId}", answer['id'] + '-year'))
+        page_spec.write(ANSWER_GETTER.replace("{answerName}", answer_name + 'Year').replace("{answerId}", answer['id'] + '-year'))
+
+    else:
```
Comment (collisdigital) [inline tests/functional/generate_pages.py]: Could this actually be `elif answer['type'] == 'TextField':` ? possible with a final `else` that throws an Exception so it's obvious when the type isn't handled.

File: app/data/schema/schema-v1.json [hunk 5]
```diff
@@ -1,258 +1,265 @@
 {
```
Comment (ajmaddaford) [inline app/data/schema/schema-v1.json:1]: The reformatting should be done in a separate commit, it's not clear what changes (if any) there are.

File: app/questionnaire/navigator.py [hunk 6]
```diff
@@ -14,10 +14,12 @@ def evaluate_rule(rule, answer):
     match_value = when['value']
     condition = when['condition']
 
+    answer_to_test = answer[0] if (isinstance(answer, list) and len(answer) == 1) else answer
```
Comment (ajmaddaford) [inline app/questionnaire/navigator.py]: I think this is superseded by the work in #578. Mentioning it here as it's done in a different place in the other pull request.

File: tests/functional/generate_pages.py [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+logger = logging.getLogger(__name__)
+logging.basicConfig(level=logging.INFO)
+
+
+SPEC_PAGE_HEADER = r"""import chai from 'chai'
+import {startQuestionnaire} from '../helpers'
+
+"""
+
+SPEC_PAGE_IMPORT = r"""import {pageName} from '../pages/surveys/census/household/{pageFile}'
```
Comment (ajmaddaford) [inline tests/functional/generate_pages.py]: Seems a bit census specific 😛

**Review by ajmaddaford — CHANGES_REQUESTED** [review]

File: tests/functional/generate_pages.py [hunk 8]
```diff
@@ -0,0 +1,209 @@
...
+def process_block(block, dir_out, spec_out):
+    logger.debug("Processing Block: %s", block['id'])
+
+    page_filename = block['id'] + '.page.js'
+    page_path = os.path.join(dir_out, page_filename)
+
+    logger.info("Creating %s...", page_path)
+
+    with open(page_path, 'w') as page_spec:
+        muntiple_choice_check = find_kv(block, 'type', ['Radio', 'Checkbox'])
```
Comment (benjefferies) [inline tests/functional/generate_pages.py]: muntiple -> multiple?

File: app/data/census_household.json [hunk 9]
```diff
@@ -0,0 +1,3478 @@
...
+                            }
+                        }
+                    ]
+                },
+                {
+                    "id": "else-permanent-or-family-home",
+                    "title": "Who lives here?",
+                    "sections": [
+                        {
+                            "id": "else-permanent-or-family-home_section",
```
Comment (benjefferies) [inline app/data/census_household.json]: _section should be -section I guess?

File: app/data/census_household.json [hunk 10]
```diff
@@ -0,0 +1,3478 @@
...
+                                                {
+                                                    "label": "Yes",
+                                                    "value": "Yes"
+                                                },
+                                                {
+                                                    "label": "No",
+                                                    "value": "No"
+                                                }
+                                            ],
+                                            "q_code": "20",
```
Comment (benjefferies) [inline app/data/census_household.json:103]: Do we know what these q_codes should be?

File: app/questionnaire/navigator.py [hunk 11]
```diff
@@ -16,6 +16,8 @@ def evaluate_rule(rule, answer_value):
     match_value = when['value']
     condition = when['condition']
 
+    # answer_to_test = answer[0] if (isinstance(answer, list) and len(answer) == 1) else answer
```
Comment (benjefferies) [inline app/questionnaire/navigator.py]: Can we delete the line if not needed?

**Review by benjefferies — CHANGES_REQUESTED** [review]

File: app/data/census_household.json (hunk 10 above)
Comment (jonnyshaw89) [reply app/data/census_household.json:103]: no not yet.

File: app/questionnaire/navigator.py (hunk 11 above)
Comment (jonnyshaw89) [reply app/questionnaire/navigator.py]: ah yes, that got left after my merge fixes

**Review by benjefferies — APPROVED** [review]

**Review by ajmaddaford — APPROVED** [review]

**Pull request merged**